src/openrct2/ride/Vehicle.cpp

```diff
@@ -7432,12 +7432,11 @@ static void vehicle_update_scenery_door(rct_vehicle* vehicle)
         trackBlock++;
     }
     const rct_track_coordinates* trackCoordinates = &TrackCoordinates[trackType];
-    int32_t x = floor2(vehicle->x, 32);
-    int32_t y = floor2(vehicle->y, 32);
-    int32_t z = (vehicle->track_z - trackBlock->z + trackCoordinates->z_end) >> 3;
+    auto wallCoords = CoordsXYZ{ CoordsXY{ vehicle->x, vehicle->y }.ToTileStart(),
+                                 vehicle->track_z - trackBlock->z + trackCoordinates->z_end };
```

Comment:
I prefer `CoordsXYZ { vehicle->x, vehicle->y, vehicle->track_z - trackBlock->z + trackCoordinates->z_end}.ToTileStart()`. If CoordsXYZ does not yet have that method, it should be added.

Reply:
Done!